nomad/fsm_test.go

```diff
@@ -244,143 +244,58 @@ func TestFSM_UpsertNode_Canonicalize_Ineligible(t *testing.T) {
 func TestFSM_UpsertNode_NodePool(t *testing.T) {
```

Comment:
I realized that I basically had the same test cases as in `TestStateStore_UpsertNode_NodePool` here, and that seem quite excessive and unnecessary, so I updated this test to only exercise the logic specific to the FSM (node canonicalization and setting the upsert options).